### PR TITLE
Added create project command and client

### PR DESCRIPTION
**Pull request by febg**
### Flags
- --name (required)
- --description (required)
- --path (optional)
    - default: `./.padlfile.yaml`
- --json (optional)
    - default: `json`

### Example
```
-> ./padl project create --name felipeproj --description "A project" --path "./.padlfile.json"

created project a2dbe0db-b1a1-4d2c-8206-138b0b139929 successfully!
```
- .padlfile.json
```
{
  "data": {
    "project_id": "3b4d0f05-0199-4b02-8169-632b9ca3cc5d",
    "variables": {
      "var1": "_var1_"
    },
    "keys": [
      "key1"
    ],
    "shared_key": "mockSharedKey"
  },
  "HMAC": "2750058f18b5ec7b6c4aea09d991742b76c325c85365b5dee51b078760265a2b634bdb6310625cc3e882600fd2dc5bafcd40a4987dcb5608187c5b6d56fc7bd3"
}
```
- yaml 
```
data:
  project_id: a2dbe0db-b1a1-4d2c-8206-138b0b139929
  variables:
    var1: _var1_
  keys:
  - key1
  shared_key: mockSharedKey
HMAC: f98e3c5e4ec14bcc7ab014938bf311c9e3cbc61f81e0b49ef7666d7c32195efe7a20ccca444428e70d70663d82f81b7bed67069876be622cb5768410c0d9ab64
```

### CODE REVIEW

File: api/payloads/project.go [hunk 1]
```diff
@@ -32,7 +36,7 @@ type DeleteDeployKeyRequest struct {
 
 // NewProjectResponse TODO
 type NewProjectResponse struct {
-	ID string `json:"id"`
+	Padfile *padlfile.File `json:"id"`
```
Comment (adrianosela) [inline api/payloads/project.go]: no need to wrap this around a `NewProjectResponse`, just marshal the padlfile directly

File: api/service/project.go [hunk 2]
```diff
@@ -40,13 +41,35 @@ func (s *Service) createProjectHandler(w http.ResponseWriter, r *http.Request) {
...
+	// Store project in db
 	if err := s.database.PutProject(project); err != nil {
 		w.WriteHeader(http.StatusBadRequest)
 		w.Write([]byte(fmt.Sprintf("could not save new project: %s", err)))
 		return
 	}
-	// marshal response
-	byt, err := json.Marshal(&payloads.NewProjectResponse{ID: project.ID})
+
+	byt, err := json.Marshal(&payloads.NewProjectResponse{Padfile: pf})
```
Comment (adrianosela) [inline api/service/project.go]: see comment above, just marshal the &pf, and return that in the client

File: cli/commands/project.go [hunk 3]
```diff
@@ -18,10 +21,61 @@ var ProjectCmds = cli.Command{
...
+func createProjectValidator(ctx *cli.Context) error {
+	return assertSet(ctx, nameFlag, descriptionFlag)
+}
+
+func createProjectHandler(ctx *cli.Context) error {
+	c, err := getClient(ctx)
+	if err != nil {
+		return fmt.Errorf("could not initialize client: %s", err)
+	}
+	p := ctx.String(name(pathFlag))
```
Comment (adrianosela) [inline cli/commands/project.go]: call it path

File: cli/commands/project.go [hunk 4]
```diff
@@ -18,10 +21,61 @@ var ProjectCmds = cli.Command{
...
+
+	if p == "" {
+		if json {
+			p = "./padlfile.json"
+		} else {
+			p = "./padlfile.yaml"
+		}
+	}
+
+	if !strings.Contains(p, ".json") && !strings.Contains(p, ".yaml") {
```
Comment (adrianosela) [inline cli/commands/project.go]: use strings.HasSuffix()

File: cli/commands/project.go [hunk 5]
```diff
@@ -18,10 +21,61 @@ var ProjectCmds = cli.Command{
...
+		return fmt.Errorf("provide file name and type")
```
Comment (adrianosela) [inline cli/commands/project.go]: `invalid file extension, must be one of { ".yaml", ".json" }`

File: cli/commands/project.go [hunk 6]
```diff
@@ -18,10 +21,61 @@ var ProjectCmds = cli.Command{
...
+	}
+
+	pf, err := c.CreateProject(pname, descr)
+
```
Comment (adrianosela) [inline cli/commands/project.go]: kill newline

**Review by adrianosela — APPROVED** [review]
Minor nit about the payload, else looks great

File: api/payloads/project.go (hunk 1 above)
Comment (febg) [reply api/payloads/project.go]: 👍 
Removed NewProjectResponse structs and return File directly

File: api/service/project.go (hunk 2 above)
Comment (febg) [reply api/service/project.go]: Same as above

File: cli/commands/project.go (hunk 3 above)
Comment (febg) [reply cli/commands/project.go]: 👍 
Added new `path` flag

File: cli/commands/project.go (hunk 4 above)
Comment (febg) [reply cli/commands/project.go]: 👍 
Replaced Contains with HasSuffix

File: cli/commands/flags.go [hunk 7]
```diff
@@ -59,6 +59,10 @@ var (
 		Name:  "json, j",
 		Usage: "print raw json -- don't pretty print",
 	}
+	path = cli.StringFlag{
```
Comment (adrianosela) [inline cli/commands/flags.go]: dont we already have one?

File: cli/commands/project.go [hunk 8]
```diff
@@ -50,7 +50,7 @@ func createProjectHandler(ctx *cli.Context) error {
 	if err != nil {
 		return fmt.Errorf("could not initialize client: %s", err)
 	}
-	p := ctx.String(name(pathFlag))
+	p := ctx.String(name(path))
```
Comment (adrianosela) [inline cli/commands/project.go]: i meant the receivinbg variable name `p` --> `path`

File: cli/commands/project.go [hunk 9]
```diff
@@ -63,12 +63,11 @@ func createProjectHandler(ctx *cli.Context) error {
 		}
 	}
 
-	if !strings.Contains(p, ".json") && !strings.Contains(p, ".yaml") {
-		return fmt.Errorf("provide file name and type")
+	if !strings.HasSuffix(p, ".json") && !strings.HasSuffix(p, ".yaml") {
+		return fmt.Errorf("invalid file extension, must be one of { \".yaml\", \".json\" }")
```
Comment (adrianosela) [inline cli/commands/project.go:67]: nice

File: cli/commands/flags.go (hunk 7 above)
Comment (febg) [reply cli/commands/flags.go]: @adrianosela I thought you meant change the name to have to separate path variables :'D

File: cli/commands/project.go (hunk 8 above)
Comment (febg) [reply cli/commands/project.go]: Gotcha!

**Review by adrianosela — APPROVED** [review]

**Comment by adrianosela** [comment]
merge at will

**Pull request merged**